core/handler/device/store.go

```diff
@@ -23,11 +24,13 @@ type Store interface {
 	DownlinkQueue(appID, devID string) (DownlinkQueue, error)
 	Set(new *Device, properties ...string) (err error)
 	Delete(appID, devID string) error
+	AttrWhitelist() map[string]bool
 }
 
 const defaultRedisPrefix = "handler"
 const redisDevicePrefix = "device"
 const redisDownlinkQueuePrefix = "downlink"
+const redisUpStreamMetadataKeyWhitelist = ":attr-whitelist"
```

Comment:
4. I see. Then prioritize the one already on the store and after take the first n till maximum allowed ?